src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -746,12 +746,17 @@ private string AppendSuffix(string parentId, string suffix, char delimiter)
             return parentId.Substring(0, trimPosition) + overflowSuffix + '#';
         }
 
-        private string GenerateRootId()
+        private static string GenerateRootId()
         {
             // It is important that the part that changes frequently be first, because
-            // many hash functions don't 'randomize' the tail of a string.   This makes
-            // sampling based on the hash produce poor samples.
-            return '|' + Interlocked.Increment(ref s_currentRootId).ToString("x") + s_uniqSuffix;
+            // some sampling functions don't sample from the high entropy part of their hash function.
+            // This makes sampling based on this produce poor samples.
+            Span<char> result = stackalloc char[1 + 16 + s_uniqSuffix.Length]; // max length needed
+            result[0] = '|';
+            bool formatted = Interlocked.Increment(ref s_currentRootId).TryFormat(result.Slice(1), out int charsWritten, "x");
```

Comment:
I think you may have undone too much... this TryFormat won't work on netfx (unless there's a polyfill extension method I'm not aware of somewhere in this library).

Reply:
😢

Reply:
Adding such an extension method into the netfx build might be a simpler / better solution than forking this method.